File: services/api/app/domain/events.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class EventType:
    RUN_STATUS: str = "run.status"
    RESEARCH_READY: str = "research.ready"
    STRATEGIES_READY: str = "strategies.ready"
    BRANCH_STARTED: str = "branch.started"
    BRANCH_PROGRESS: str = "branch.progress"
    BRANCH_COMPLETED: str = "branch.completed"
    WINNER_SELECTED: str = "winner.selected"
    RUN_COMPLETED: str = "run.completed"
    RUN_FAILED: str = "run.failed"


EVENTS = EventType()

File: apps/web/src/lib/events.ts
import type { RunEvent } from './api'

export const EVENT_TYPES = [
  'run.status',
  'research.ready',
  'strategies.ready',
  'branch.started',
  'branch.progress',
  'branch.completed',
  'winner.selected',
  'run.completed',
  'run.failed',
] as const

export function parseRunEvent(raw: MessageEvent<string>): RunEvent | null {
  try {
    return JSON.parse(raw.data) as RunEvent
  } catch {
    return null
  }
}

